num/quat/exp.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quat
+
+import "math"
+
+// Exp returns e**q, the base-e exponential of q.
+func Exp(q Quat) Quat {
+	w, uv := split(q)
+	if uv == zero {
+		return lift(math.Exp(w))
+	}
+	v := Abs(uv)
+	e := math.Exp(w)
+	s, c := math.Sincos(v)
+	return Add(lift(e*c), Scale(e*s/v, uv))
```

Comment:
Instead of these `Add`, I think it would be clearer to have a `pack` function that is the opposite of `split`. It makes it clearer that the relevant fields are explicitly zero.

Reply:
Do we need to make that clearer? The expressions here reflect the maths more closely. I have added this is a separate commit.